merlin/models/tf/core/combinators.py

```diff
@@ -453,6 +453,56 @@ def select_by_names(self, names: List[str]) -> Optional[List[Block]]:
                 raise ValueError(f"Given name {name} is not in ParallelBlock {self.name}")
         return blocks
 
+    def select_by_tag(
+        self,
+        tags: Union[str, Tags, List[Union[str, Tags]]],
+    ) -> Optional["ParallelBlock"]:
+        """Select layers of parallel blocks by tags.
+
+        This method will return a ParallelBlock instance with all the branches that
+        have at least one feature that matches any of the tags provided.
+
+        For example, this method can be useful when a ParallelBlock has both item and
+        user features in a two-tower model or DLRM, and we want to select only the item
+        or user features.
+
+        >>> all_inputs = InputBlockV2(schema)  # InputBlock is also a ParallelBlock
+        >>> item_inputs = all_inputs.select_by_tag(Tags.ITEM)
+        ['continuous', 'embeddings']
+        >>> item_inputs.schema["continuous"].column_names
+        ['item_recency']
+        >>> item_inputs.schema["embeddings"].column_names
+        ['item_id', 'item_category', 'item_genres']
+
+        Parameters
+        ----------
+        tags: str or Tags or List[Union[str, Tags]]
+             List of tags that describe which blocks to match
+
+        Returns
+        -------
+        ParallelBlock
+        """
+        if not isinstance(tags, (list, tuple)):
+            tags = [tags]
+
+        selected_branches = {}
+        selected_schemas = Schema()
+
+        for name, branch in self.parallel_dict.items():
+            branch_has_schema = getattr(branch, "has_schema", False)
+            if not branch_has_schema:
+                continue
+            selected_branch = branch.select_by_tag(tags)
+            if not selected_branch:
+                continue
+            selected_branches[name] = selected_branch
+            selected_schemas += selected_branch.schema
+
+        if not selected_branches:
+            return
```

Comment:
I am wondering what the most logical behaviour here is? I am inclinded to say that maybe raising an exception could be better. Thoughts @gabrielspmoreira @oliverholworthy @edknv?

Reply:
I returned `None` here because I was following the behavior of `select_by_name` which returns `None` when there is no match. I'm not sure which is better, but I think we should be consistent in the behavior of `select_by` methods.

Reply:
@marcromeyn I tried raising an error here, and it looks like we will need `try... except` blocks and maybe even a custom exception because some blocks are themselves `ParallelBlock`s, e.g., when a branch is an `Embeddings` which itself is a `ParallelBlock`. From the user perspective, it's probably better to raise an exception, but from the implementation perspective, I think just returning nothing here is okay.